### PR TITLE
fix: discord webhook error

### PR DESCRIPTION
**Pull request by anhnh12**
**What does this PR do?**
-   [x] Fix webhook 500 error: because some guilds were not created when Mochi Bot joined (maybe server was down at that time)
- [x] Remove guild_id fkey form guild_config_activities and guild_user_activity_logs

### CODE REVIEW

File: pkg/job/fetch_discord_users.go [hunk 1]
```diff
@@ -25,6 +25,13 @@ func (j *fetchDiscordUsers) Run() error {
 		return err
 	}
 
+	// fetch current guilds from discord rest API
```
Comment (trkhoi) [inline pkg/job/fetch_discord_users.go:28]: remove cmt

File: pkg/entities/guild.go [hunk 2]
```diff
@@ -241,3 +241,92 @@ func (e *Entity) GetGuildRoles(guildID string) (*response.DiscordGuildRoles, err
...
+		result = append(result, dbGuild)
+	}
+
+	return result, nil
+}
+
+func (e *Entity) CreateGuildIfNotExists(guildID string) error {
+	l := e.log.Fields(logger.Fields{"guildID": guildID})
+
+	_, err := e.repo.DiscordGuilds.GetByID(guildID)
```
Comment (trkhoi) [inline pkg/entities/guild.go:305]: chỗ này mình dùng hàm FirstOrCreate của gorm đc ko a

File: pkg/entities/guild.go [hunk 3]
```diff
@@ -241,3 +241,92 @@ func (e *Entity) GetGuildRoles(guildID string) (*response.DiscordGuildRoles, err
...
+	}
+
+	dbGuilds := make(map[string]*response.GetGuildResponse)
+	for _, g := range res.Data {
+		dbGuilds[g.ID] = g
+	}
+
+	fetchedGuilds := make(map[string]*discordgo.Guild)
+	e.discord.StateEnabled = true
+	fmt.Println("e.discord.State.Guilds:", e.discord.State.Guilds)
```
Comment (trkhoi) [inline pkg/entities/guild.go:259]: remoev log

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/entities/guild.go (hunk 2 above)
Comment (anhnh12) [reply pkg/entities/guild.go:305]: k dc do k co san guild data

File: pkg/entities/guild.go [hunk 4]
```diff
@@ -241,3 +241,92 @@ func (e *Entity) GetGuildRoles(guildID string) (*response.DiscordGuildRoles, err
...
+	for _, g := range e.discord.State.Guilds {
+		fetchedGuilds[g.ID] = g
+	}
+
+	result := make([]*response.GetGuildResponse, 0)
+
+	for id, g := range dbGuilds {
+		_, ok := fetchedGuilds[id]
+		if !ok {
```
Comment (vhbien000) [inline pkg/entities/guild.go:268]: nhìn nó bằng phẳng hơn thôi chứ kcg
```suggestion
		if ok {
		   result = append(result, g)
		   continue 
		}
		bla bla
```

File: pkg/entities/guild.go [hunk 5]
```diff
@@ -241,3 +241,92 @@ func (e *Entity) GetGuildRoles(guildID string) (*response.DiscordGuildRoles, err
...
+			}
+			continue
+		}
+
+		result = append(result, g)
+	}
+
+	for id, g := range fetchedGuilds {
+		dbGuild, ok := dbGuilds[id]
+		if !ok || !dbGuild.Active {
```
Comment (vhbien000) [inline pkg/entities/guild.go:285]: giống bên trên
```suggestion
		if ok && dbGuild.Active {
		   result = append(result, dbGuild)
		   continue
		}
		
		err = e.CreateGuild(request.CreateGuildRequest{
			ID:       id,
			Name:     g.Name,
			JoinedAt: g.JoinedAt,
		})
		if err != nil {
			e.log.Fields(logger.Fields{"guildID": id}).Error(err, "[entity.FetchAndSyncGuilds] DeactivateGuild() failed")
			}
```

File: pkg/entities/webhook.go [hunk 6]
```diff
@@ -726,3 +726,23 @@ func (e *Entity) sendProposalToDiscord(wg *sync.WaitGroup, channelId string, pro
 	defer wg.Done()
 	e.svc.Discord.NotifyNewProposal(channelId, *proposal)
 }
+
+func (e *Entity) HandleGuildCreate(guildID string) error {
+	l := e.log.Fields(logger.Fields{"guildID": guildID})
+
+	if err := e.InitGuildDefaultTokenConfigs(guildID); err != nil {
+		l.Error(err, "[entity.handleGuildCreate] InitGuildDefaultTokenConfigs() failed")
+	}
```
Comment (vhbien000) [inline pkg/entities/webhook.go:735]: nitpick: line break

File: pkg/repo/discord_guilds/pg.go [hunk 7]
```diff
@@ -29,12 +29,16 @@ func (pg *pg) Gets() ([]model.DiscordGuild, error) {
 }
 
 func (pg *pg) CreateOrReactivate(guild model.DiscordGuild) error {
+	joinedAt := guild.JoinedAt
+	if !guild.JoinedAt.IsZero() {
+		joinedAt = time.Now().UTC()
+	}
```
Comment (vhbien000) [inline pkg/repo/discord_guilds/pg.go:35]: nitpick: line break

**Review by vhbien000 — APPROVED** [review]

**Review by trkhoi — APPROVED** [review]

**Pull request merged**